sdks/python/apache_beam/ml/transforms/embeddings/open_ai.py

```diff
@@ -0,0 +1,223 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# OpenAI Python SDK is required for this module.
+# Install using: pip install openai
+
+import logging
+import time
+from collections.abc import Iterable
+from collections.abc import Sequence
+from typing import Any, Optional, TypeVar, Union, Tuple, cast
+
+import apache_beam as beam
+from apache_beam.pvalue import PCollection, Row
+from apache_beam.io.components.adaptive_throttler import AdaptiveThrottler
+from apache_beam.metrics.metric import Metrics
+from apache_beam.ml.inference.base import ModelHandler
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.ml.transforms.base import EmbeddingsManager
+from apache_beam.ml.transforms.base import _TextEmbeddingHandler
+from apache_beam.utils import retry
+
+__all__ = ["OpenAITextEmbeddings"]
+
+# Define a type variable for the output
+MLTransformOutputT = TypeVar('MLTransformOutputT')
+
+# Default batch size for OpenAI calls
+_BATCH_SIZE = 20  # OpenAI can handle larger batches than Vertex
```

Comment:
This could likely be handled in the model handler instead of being hard-coded here